layers/state_tracker/shader_instruction.cpp

```diff
@@ -17,7 +17,7 @@
 #include "state_tracker/shader_module.h"
 #include "generated/spirv_grammar_helper.h"
 
-Instruction::Instruction(std::vector<uint32_t>::const_iterator it) : result_id_(0), type_id_(0) {
+Instruction::Instruction(std::vector<uint32_t>::const_iterator it) : result_id_index_(0), type_id_index_(0) {
```

Comment:
This can be handled in the header file instead.

```suggestion
Instruction::Instruction(std::vector<uint32_t>::const_iterator it) {
```

Reply:
I guess we have been doing these init at construction time everywhere else in the code, wasn't sure "when" it should be done in the header vs here

Reply:
Less chance of bugs if someone makes another constructor when doing it in the header file in this instance.